tests/test_flows.py

```diff
@@ -436,36 +436,83 @@ def test_channel_selection(self):
             "{0x82 {0x00 0x00}}"
             )
 
-        debug("Send channel selection request")
-        mid = self.gateway_ucc.dev_send_1905(
-                self.mac_repeater1,
-                0x8006,
-                tlv(0x8B, 0x005F, "{} {}".format(self.mac_repeater1_wlan0, payload_wlan0)),
-                tlv(0x8D, 0x0007, "{} {}".format(self.mac_repeater1_wlan0, payload_empty)),
-                tlv(0x8B, 0x004C, "{} {}".format(self.mac_repeater1_wlan2, payload_wlan2)),
-                tlv(0x8D, 0x0007, "{} {}".format(self.mac_repeater1_wlan2, payload_empty))
+        """
+        Step 1: Trigger channel selection to channel 6 and 36. Check that
+                operating channel report was sent.
+
+        Step 2: Trigger channel selection to channel 6 and 36 again - check that
+                operating channel report is sent again. This is to catch bugs
+                when we don't send channel report when there is no need to
+                switch channel
+        """
+        for i in range(1, 3):
+            debug("Send channel selection request, step {}".format(i))
+            mid = self.gateway_ucc.dev_send_1905(
+                    self.mac_repeater1,
+                    0x8006,
+                    tlv(0x8B, 0x005F, '{} {}'.format(self.mac_repeater1_wlan0, payload_wlan0)),
+                    tlv(0x8D, 0x0007, '{} 0x{:2x}'.format(self.mac_repeater1_wlan0, tp20dBm)),
+                    tlv(0x8B, 0x004C, '{} {}'.format(self.mac_repeater1_wlan0, payload_wlan2)),
+                    tlv(0x8D, 0x0007, '{} 0x{:2x}'.format(self.mac_repeater1_wlan2, tp20dBm))
             )
-        time.sleep(1)
+            time.sleep(1)
+
+            debug(
+                "Confirming channel selection request has been received on controller,\
+                    step {}".format(i))
+            self.check_log(
+                self.gateway,
+                "controller",
+                r"CHANNEL_SELECTION_RESPONSE_MESSAGE, mid={}".format(mid))
+
+            debug(
+                "Confirming channel selection request has been received on agent,step {}".format(i))
+
+            self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_SELECTION_REQUEST_MESSAGE")
+            self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_SELECTION_REQUEST_MESSAGE")
+
+            debug(
+                "Confirming tlvTransmitPowerLimit has been received with correct value on \
+                    agent, step {}".format(i))
+
+            self.check_log(
+                self.repeater1, "agent_wlan0", "tlvTransmitPowerLimit {}".format(tp20dBm))
```

Comment:
Here as well: we prefer line splitting like this.
```suggestion
            self.check_log(self.repeater1, 
                           "agent_wlan0", "tlvTransmitPowerLimit {}".format(tp20dBm))
```

Reply:
In some places, we still can get lines greater than 100 symbols. And the solution - transfer all parameters to the new string including first.